lib/openstack/openstack_handle/handle.rb

```diff
@@ -60,9 +63,15 @@ def self.raw_connect(username, password, auth_url, service = "Compute", extra_op
       raise
     end
 
-    def self.auth_url(address, port = 5000, ssl = false)
-      scheme = ssl ? "https" : "http"
-      "#{scheme}://#{address}:#{port}/v2.0/tokens"
+    def self.auth_url(address, port = 5000, scheme = "http")
+      url(address, port, scheme)
+    end
+
+    def self.url(address, port = 5000, scheme = "http", path = "/v2.0/tokens")
+      port = port.to_i
```

Comment:
Ok, that's fine, I suppose someone could call this with a string in other places in addition to automate.